### PR TITLE
NUnit 3.x Upgrade

### PR DESCRIPTION
**Pull request by tomspilman**
This upgrades our unit tests to use NUnit 3.9.0 and the NUnitLite console runner.

I fixed most all the tests to work with NUnit 3.x, but a few that were trickier i just `[Ignore]`d for now.  I also ensured our XNA tests build, but there are some broken tests that look to have been broken for a while.  We can address this later when the XNA tests are added to the automated build.

With this upgrade we now have 1117 of our 1227 unit tests running on Mac (via the DesktopGL platform which is technically the Linux build of MG).

Note we are resolving the NUnit dependencies via NuGet instead of our dependencies repo.  We'll see how this goes for us and if we run into any issues with it.

Also as of this PR our unit tests work within Visual Studio Test Explorer if you have the NUnit extension installed.

Fixes https://github.com/MonoGame/MonoGame/issues/5255
Fixes https://github.com/MonoGame/MonoGame/issues/5254

### CODE REVIEW

**Comment by tomspilman** [comment]
Back to the issue i had months ago first time we tried running the unit tests on Mac.  After the tests finish and the report is written to disk the test runner hangs and never exits the process.  Don't know why yet.

**Comment by tomspilman** [comment]
So i found these two tests both pass on Mac, but subsequently keep the test runner from shutting down correctly:
```
MonoGame.Tests.Graphics.RenderTarget2DTest.ZeroSizeShouldFailTest
MonoGame.Tests.Graphics.RenderTargetCubeTest.ZeroSizeShouldFailTest
```
You can test them individually by doing:
```
mono MonoGameTests.exe --test=MonoGame.Tests.Graphics.RenderTarget2DTest.ZeroSizeShouldFailTest
mono MonoGameTests.exe --test=MonoGame.Tests.Graphics.RenderTargetCubeTest.ZeroSizeShouldFailTest
```
I disabled them for now, but someone with more Mac/OpenGL experience should try to figure this out.

**Comment by tomspilman** [comment]
Yay...  Mac tests are now working... at least most all of them.
image

**Comment by tomspilman** [comment]
@dellis1972 - This is good to merge, but maybe review the changes first.

File: Test/ContentPipeline/FbxImporterTests.cs [hunk 1]
```diff
@@ -150,21 +153,21 @@ public void Dude()
...
-            Paths.AreEqual(@"Assets\Models\Dude\pantsN.tga", materials["character_anim:pantsM"].Textures["Bump"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upBodyC.tga", materials["character_anim:upBodyM"].Textures["Texture"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upBodyS.tga", materials["character_anim:upBodyM"].Textures["Specular"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upbodyN.tga", materials["character_anim:upBodyM"].Textures["Bump"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upBodyC.tga", materials["character_anim:eyeBallM"].Textures["Texture"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upBodyS.tga", materials["character_anim:eyeBallM"].Textures["Specular"].Filename);
-            Paths.AreEqual(@"Assets\Models\Dude\upbodyN.tga", materials["character_anim:eyeBallM"].Textures["Bump"].Filename);
+            Paths.AreEqual(@"Assets/Models/Dude/head.tga", materials["character_anim:headM"].Textures["Texture"].Filename);
+            Paths.AreEqual(@"Assets/Models/Dude/headS.tga", materials["character_anim:headM"].Textures["Specular"].Filename);
```
Comment (dellis1972) [inline Test/ContentPipeline/FbxImporterTests.cs]: how does this work on Windows? Does it produce `/` in our imported there too?

File: Test/Framework/GameComponentTest.cs [hunk 2]
```diff
@@ -11,6 +11,9 @@ namespace MonoGame.Tests.Framework
     public static class GameComponentTest
     {
         [Test]
+#if DESKTOPGL
+        [Ignore("This crashes inside SDL on Mac!")]
+#endif
```
Comment (dellis1972) [inline Test/Framework/GameComponentTest.cs]: @cra0zy we should try and figure out why :)

**Comment by dellis1972** [comment]
@tomspilman looks ok a couple of questions in the comments though. We should probably open issues for each of those `Ignore this crashes on XXXX` so we don't loose sight of them

File: Test/ContentPipeline/FbxImporterTests.cs (hunk 1 above)
Comment (tomspilman) [reply Test/ContentPipeline/FbxImporterTests.cs]: If you look at my Paths.AreEqual you'll see it normalizes the slashes via Path.GetFullPath.  However it doesn't work on Mac without me flipping these.

So it works.

**Comment by tomspilman** [comment]
I figured i would open an issue with the full list of skipped tests on Mac to address.  We also have some broken/disabled tests on Windows which i plan to open an issue for as well.

**Comment by dellis1972** [comment]
ok looks good in that case. I'll merge

**Comment by tomspilman** [comment]
FYI. https://github.com/MonoGame/MonoGame/issues/6253

**Pull request merged**